src/zeta/aca_zeta_programming.cpp

```diff
@@ -116,40 +71,38 @@ int ACA_Zeta_Programming::delete_zeta_config(const alcor::schema::AuxGateway cur
   zeta_config stZetaCfg;
   int overall_rc = EXIT_SUCCESS;
 
-  uint group_id = get_or_create_group_id(current_AuxGateway.id());
-  stZetaCfg.group_id = to_string(group_id);
-
-  string auxGateway_id = current_AuxGateway.id();
-
-  for (auto destination : current_AuxGateway.destinations()) {
-    stZetaCfg.zeta_buckets.push_back(destination.ip_address());
-  }
-  uint oam_port = current_AuxGateway.zeta_info().port_inband_operation();
-
-  // Reset auxGateway_id to empty
-  ACA_Vlan_Manager::get_instance().set_aux_gateway(tunnel_id, "");
+  auxgateway_entry auxGateway =
+          ACA_Vlan_Manager::get_instance().get_auxgateway_unsafe(tunnel_id);
+  stZetaCfg.group_id = auxGateway.group_id;
 
-  if (!ACA_Vlan_Manager::get_instance().is_exist_aux_gateway(auxGateway_id)) {
-    // update oam_ports_cache and delete the OAM punt rule
-    Aca_Oam_Port_Manager::get_instance().remove_oam_port_rule(oam_port);
-    // delete the group bucket rule
-    overall_rc = _delete_zeta_group_entry(&stZetaCfg);
+  if (auxGateway.auxGateway_id.empty()) {
+    ACA_LOG_INFO("%s", "No auxgateway is currently set for this vpc!\n");
+  } else {
+    if (auxGateway.auxGateway_id != current_AuxGateway.id()) {
+      ACA_LOG_ERROR("%s", "The auxgateway_id is inconsistent with the auxgateway_id currently set by the vpc!\n");
+    } else {
+      ACA_LOG_INFO("%s", "Reset auxGateway to empty!\n");
+      overall_rc = ACA_Vlan_Manager::get_instance().remove_auxgateway(tunnel_id);
+    }
   }
 
   return overall_rc;
 }
 
-int ACA_Zeta_Programming::_create_or_update_zeta_group_entry(zeta_config *zeta_cfg)
+int ACA_Zeta_Programming::_create_zeta_group_entry(zeta_config *zeta_cfg)
 {
   unsigned long not_care_culminative_time;
   int overall_rc = EXIT_SUCCESS;
-
+  if (is_exist_group_rule(zeta_cfg->group_id)) {
+    return overall_rc;
+  }
   //adding group table rule
-  string cmd = "-O OpenFlow13 add-group br-tun group_id=" + zeta_cfg->group_id + ",type=select";
+  string cmd = "-O OpenFlow13 add-group br-tun group_id=" + to_string(zeta_cfg->group_id) +
```

Comment:
please confirm the group rule can send both regular packets and also ARP packets to the Zeta gateway.